Add jump forward feature, move all code/libraries out of FLA to allow grunt build.

stevemayhew opened the pull request:
- The FlashMediaElement.fla now matches the FlashMediaElement.swc.  With swfdump you can see it has no ABC (grunt build compiles all the AS3 code).
- Simplified the CDN build path using a flash config conditional 
- Added a jump forward feature to mirror the skip back.
- added an error handler to log Flashls errors in debug mode.

johndyer commented:
Oh wow, that CDN build is way less hacky. Good stuff.

johndyer commented:
I've got this all merged in, but I'm having one problem. When I build with grunt, I get the following error: `Error: unable to open 'libs/player/10.0/playerglobal.swc`

Does that ring a bell? I can't seem to find anything that references that.

johndyer commented:
Whoops. Nevermind, I'll download the 10.0 version.

johndyer commented:
@stevemayhew Did anything happen in the *.fla file to make it include additional output text? When I compile using your new .swc file, I see the following output: 
![image](https://cloud.githubusercontent.com/assets/214227/9023582/9b90c1f6-3856-11e5-9932-cd35db8c965b.png)

stevemayhew commented:
Hey John,
Yeah I had a bunch of trouble with matching the fla / flashmediaelement.swc versions from mangui’s changes…  I wanted to make it easier to use a new flashls.swc but did not get very far with it.  I think what we really want to do is build the fullscreen controls from the images in the HTML player and ditch having to use the adobe tools completely (most people that contribute to open source don’t have a subscription to Adobe Creative Cloud ;-)).

What do you think?

PSS I’ve not finished skip forward/back for the fullscreen yet.
-s

I think when Mangui 

> On Aug 1, 2015, at 2:07 PM, John Dyer <notifications@github.com> wrote:
> 
> @stevemayhew <https://github.com/stevemayhew> Did anything happen in the *.fla file to make it include additional output text? When I compile using your new .swc file, I see the following output: 
>  <https://cloud.githubusercontent.com/assets/214227/9023582/9b90c1f6-3856-11e5-9932-cd35db8c965b.png>
> —
> Reply to this email directly or view it on GitHub <https://github.com/johndyer/mediaelement/pull/1480#issuecomment-126953660>.
>

johndyer commented:
Makes sense. What I'd like to do is simplify Flash by removing all controls
and just using the HTML controls going forward. They were there for IE6-8
and for Firefox back when you couldn't mix browser-based fullscreen and
Flash.

Do you know what's causing the output of types in the attached image above?

JD

On Sat, Aug 1, 2015 at 4:38 PM, Steve Mayhew <notifications@github.com>
wrote:

> Hey John,
> Yeah I had a bunch of trouble with matching the fla /
> flashmediaelement.swc versions from mangui’s changes… I wanted to make it
> easier to use a new flashls.swc but did not get very far with it. I think
> what we really want to do is build the fullscreen controls from the images
> in the HTML player and ditch having to use the adobe tools completely (most
> people that contribute to open source don’t have a subscription to Adobe
> Creative Cloud ;-)).
>
> What do you think?
>
> PSS I’ve not finished skip forward/back for the fullscreen yet.
> -s
>
> I think when Mangui
>
> > On Aug 1, 2015, at 2:07 PM, John Dyer <notifications@github.com> wrote:
> >
> > @stevemayhew <https://github.com/stevemayhew> Did anything happen in
> the *.fla file to make it include additional output text? When I compile
> using your new .swc file, I see the following output:
> > <
> https://cloud.githubusercontent.com/assets/214227/9023582/9b90c1f6-3856-11e5-9932-cd35db8c965b.png
> >
> > —
> > Reply to this email directly or view it on GitHub <
> https://github.com/johndyer/mediaelement/pull/1480#issuecomment-126953660
> >.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/johndyer/mediaelement/pull/1480#issuecomment-126955488>
> .
>



-- 
John Dyer - http://j.hn/

stevemayhew commented:
I can't see the screen shot.  Are you compiling with Grunt?  Can you send the text of what you get as output?

Here's the sha1 on the swc file I have:

```
$ shasum src/flash/flashmediaelement.swc 
5a1f7650aa5abb3441262dcbcb3c77912fe9b13f  src/flash/flashmediaelement.swc
```
Note the only AS3 class that is defined in the SWC now is the FlashMediaElementDisplay:

```
$ swfdump  src/flash/flashmediaelement.swc  | grep -i class
  <SymbolClass>
    <Symbol idref='32' className='FlashMediaElementDisplay' />
  </SymbolClass>
```

This is the class that has all the sprites from the buttons, controls etc that are loaded by the code in FlashMediaElement (e.g. _mediaElementDisplay.getChildByName("fullscreen_btn")).  Next step would be to create directly from the images in the web side (rather then using Adobe Flash Professional with the FLA file).

The rest of the AS3 source is included by the Grunt build command:

```
        flexPath: '../flex_sdk_4.6',
        buildFlashCommand: [
            '<%= flexPath %>/bin/mxmlc -strict=false -compiler.debug -warnings=true',
            'src/flash/FlashMediaElement.as -o <%= flashOut %>',
            '-define+=CONFIG::cdnBuild,<%= cdnBuild %>',
            '-library-path+="<%= flexPath %>/lib"',
            '-include-libraries+=src/flash/flashmediaelement.swc',
            '-include-libraries+=src/flash/flashls.swc -use-network=true',
            '-source-path src/flash',
            '-target-player 10.0',
            '-headless-server -static-link-runtime-shared-libraries'
```

The -source-path does that.

Not sure if/how that has broken any workflow you have for building.

johndyer commented:
If you download the current source code, compile using Grunt, then load the /tests/test.html page, and set the JavaScript option "enableDebug: true" you'll see the output on the audio player's flash output. I'm attaching another screenshot here on Github which you can't see in the email, but can see on the site (https://github.com/johndyer/mediaelement/pull/1480)

![image](https://cloud.githubusercontent.com/assets/214227/9064188/fa2e4ce2-3a90-11e5-8d11-a2c09aec6217.png)

https://cloud.githubusercontent.com/assets/214227/9064188/fa2e4ce2-3a90-11e5-8d11-a2c09aec6217.png

stevemayhew commented:
John,
Here’s what I see in the unit test page: https://sm-share.s3.amazonaws.com/2015-08-04_1612.png <https://sm-share.s3.amazonaws.com/2015-08-04_1612.png>

Looks like some of the test media is missing from media folder (big buck bunny stuff and an MP3).   I copied an MP3 of mine and it played fine

I did see this build error:  
  Running "shell:buildFlash" (shell) task
Loading configuration file /Users/smayhew/flex_sdk_4/frameworks/flex-config.xml
/Users/smayhew/opensource/mediaelement/src/flash/htmlelements/DailyMotionElement.as(20): col: 15 Error: Interface method seekLimit in namespace htmlelements:IMediaElement not implemented by class htmlelements:DailyMotionElement.

	public class DailyMotionElement extends Sprite implements IMediaElement
	             ^

Warning: Command failed: /Users/smayhew/opensource/mediaelement/src/flash/htmlelements/DailyMotionElement.as(20): col: 15 Error: Interface method seekLimit in namespace htmlelements:IMediaElement not implemented by class htmlelements:DailyMotionElement.

	public class DailyMotionElement extends Sprite implements IMediaElement
	             ^

 Use --force to continue.

Aborted due to warnings.
stevemayhewmbpro15:mediaelement smayhew$ rm src/flash/htmlelements/DailyMotionElement.as 

I just removed the DailyMotionElement code and all is fine.  Need to implement the IMediaElement.seekLimit() I added this to fix the scroll bar click past the buffered region issue, https://github.com/johndyer/mediaelement/issues/1453 <https://github.com/johndyer/mediaelement/issues/1453>



> On Aug 4, 2015, at 8:10 AM, John Dyer <notifications@github.com> wrote:
> 
> enableDebug: true

johndyer commented:
Do you have the latest commits? I fixed the DailyMotion error in
commit f16db9995dee67de1717e0f6ebc539ccdea9dd7b

On the test page, make sure enableDebug: true and that you're pointed to
the latest flash (in local-build or whereever you have it)

JD

On Tue, Aug 4, 2015 at 6:53 PM, Steve Mayhew <notifications@github.com>
wrote:

> John,
> Here’s what I see in the unit test page:
> https://sm-share.s3.amazonaws.com/2015-08-04_1612.png <
> https://sm-share.s3.amazonaws.com/2015-08-04_1612.png>
>
> Looks like some of the test media is missing from media folder (big buck
> bunny stuff and an MP3). I copied an MP3 of mine and it played fine
>
> I did see this build error:
> Running "shell:buildFlash" (shell) task
> Loading configuration file
> /Users/smayhew/flex_sdk_4/frameworks/flex-config.xml
> /Users/smayhew/opensource/mediaelement/src/flash/htmlelements/DailyMotionElement.as(20):
> col: 15 Error: Interface method seekLimit in namespace
> htmlelements:IMediaElement not implemented by class
> htmlelements:DailyMotionElement.
>
> public class DailyMotionElement extends Sprite implements IMediaElement
> ^
>
> Warning: Command failed:
> /Users/smayhew/opensource/mediaelement/src/flash/htmlelements/DailyMotionElement.as(20):
> col: 15 Error: Interface method seekLimit in namespace
> htmlelements:IMediaElement not implemented by class
> htmlelements:DailyMotionElement.
>
> public class DailyMotionElement extends Sprite implements IMediaElement
> ^
>
> Use --force to continue.
>
> Aborted due to warnings.
> stevemayhewmbpro15:mediaelement smayhew$ rm
> src/flash/htmlelements/DailyMotionElement.as
>
> I just removed the DailyMotionElement code and all is fine. Need to
> implement the IMediaElement.seekLimit() I added this to fix the scroll bar
> click past the buffered region issue,
> https://github.com/johndyer/mediaelement/issues/1453 <
> https://github.com/johndyer/mediaelement/issues/1453>
>
>
>
> > On Aug 4, 2015, at 8:10 AM, John Dyer <notifications@github.com> wrote:
> >
> > enableDebug: true
>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/johndyer/mediaelement/pull/1480#issuecomment-127799058>
> .
>



-- 
John Dyer - http://j.hn/
